Marlin/ultralcd.cpp

```diff
@@ -1083,6 +1092,51 @@ void lcd_cooldown() {
 
 /**
  *
+ * "Change Filament" submenu
+ *
+ */
+static void lcd_change_filament_menu() {
+  START_MENU();
+  MENU_ITEM(back, MSG_MAIN);
+  MENU_ITEM(function, MSG_REMOVE_FILAMENT , lcd_retract_filament);
+  MENU_ITEM(submenu, MSG_INSERT_FILAMENT , lcd_insert_filament_menu);
+  END_MENU();
+}
+
+static void lcd_insert_filament_menu() {
+  START_MENU();
+  MENU_ITEM(back, MSG_CHANGE_FILAMENT);
+  MENU_ITEM_EDIT_CALLBACK(bool, MSG_INSERT_FILAMENT_SLOW ,&filamentSlowInsert ,lcd_insert_filament_slow);
+  if(!filamentSlowInsert)
+    MENU_ITEM(function, MSG_INSERT_FILAMENT  , lcd_insert_filament);
+  END_MENU();
+}
+
+static void lcd_insert_filament_slow(){
+ while (filamentSlowInsert) {
```

Comment:
Ah, I see. I had been thinking of the slow-insert as being useful, after initial quick insertion, to purge filament. But a slow option to align the filament with the nozzle and get the gear to catch it is also useful.

I also think the SLOW option should be available for eject, actually. This can work a lot better for a cooled-down (~170°) extruder, because at first the filament just stretches before it starts to eject from the hot end. The nozzle will be cooled down sometimes, when users want to do a "cold pull" to clean the nozzle or see the shape of the interior in the filament they pull out.

On my own printer I just use SD Card files for eject or insert. This has some advantages, such as using `M117` to print the number of millimeters remaining. The MFC feature should do something like that too, so you know how much is left to go.

The other thing my GCode does is start the eject slow and then it gradually speeds up. For a direct-drive extruder, this is less important, and it doesn't end up moving very fast. With a Bowden, it would be possible to eject a nozzle's length of filament at one speed, then the rest of the filament at maximum speed.

I have some thoughts on the speed of "slow" where it might be based on the user's settings. The speed you're setting is based on percentage, but I think it should probably be based on mm/s instead. In other words, first figure out what the printer's 100% extruder speed actually comes out to be in mm/s. Then use that to figure out the exact percentage needed to get a slow speed like 0.5mm/s or 1mm/s. I can post a code example later to make it clearer, but I've got to run right now!